nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/FetchS3Object.java

```diff
@@ -165,14 +165,19 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
             final ObjectMetadata metadata = s3Object.getObjectMetadata();
             if (metadata.getContentDisposition() != null) {
-                final String fullyQualified = metadata.getContentDisposition();
-                final int lastSlash = fullyQualified.lastIndexOf("/");
-                if (lastSlash > -1 && lastSlash < fullyQualified.length() - 1) {
-                    attributes.put(CoreAttributes.PATH.key(), fullyQualified.substring(0, lastSlash));
-                    attributes.put(CoreAttributes.ABSOLUTE_PATH.key(), fullyQualified);
-                    attributes.put(CoreAttributes.FILENAME.key(), fullyQualified.substring(lastSlash + 1));
+                final String contentDisposition = metadata.getContentDisposition();
+
+                if (contentDisposition.equals(PutS3Object.CONTENT_DISPOSITION_INLINE) || contentDisposition.startsWith("attachment; filename=")) {
+                    attributes.put(CoreAttributes.FILENAME.key(), key);
```

Comment:
Setting the`filename` to the full object key (eg. `dir1/dir2/file1`) is a bit inconsistent because `filename` typically does not contain the "path" (like in the else branch below and in case of other processor too).
I would suggest to use the simple file name:
- `dir1/dir2/file1` => `file1`
- `file2` => `file2`

Reply:
@turcsanyip Thanks for your explanation, it is clear and helps me a lot. I have updated the unit test and attribute setting code in FetchS3Object processor.